### PR TITLE
Refactor projects 01 through 04.

### PR DESCRIPTION
**Pull request by at-ninja**
For this refactor, I've split the projects codebase out from the `SETUP_APPLICATION` macro that defines how the project is loaded.

### CODE REVIEW

File: projects/01_triangle/Triangle.cpp [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+#include "ppx/ppx.h"
+using namespace ppx;
+
+#if defined(USE_DX12)
+const grfx::Api kApi = grfx::API_DX_12_0;
+#elif defined(USE_VK)
+const grfx::Api kApi = grfx::API_VK_1_1;
+#endif
+
+void TriangleApp::Config(ppx::ApplicationSettings& settings)
```
Comment (apazylbe) [inline projects/01_triangle/Triangle.cpp:25]: Curious, was there a reason ppx:: here was left when others were removed?

**Review by apazylbe — COMMENTED** [review]
Overall lgtm, just a question about 1 thing

**Review by Keenuts — APPROVED** [review]
LGTM (just the ppx nits Aliaa added)

**Comment by Keenuts** [comment]
Just one thing in case: If the end goal is just to split the `main.cc` from the rest of the code, fine. But I'd be against having a shared file for those samples:

IMO the advantage of those samples are they are simple, and easy to follow/read to discover how BigWheels works. If we start sharing code, etc, it's not a single file you have to read but fish around some shared code.

File: projects/01_triangle/Triangle.cpp [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    {
+        // clang-format off
+        std::vector<float> vertexData = {
+            // position           // vertex colors
+             0.0f,  0.5f, 0.0f,   1.0f, 0.0f, 0.0f,
+            -0.5f, -0.5f, 0.0f,   0.0f, 1.0f, 0.0f,
+             0.5f, -0.5f, 0.0f,   0.0f, 0.0f, 1.0f,
+        };
+        // clang-format on
+        uint32_t dataSize = ppx::SizeInBytesU32(vertexData);
```
[the remark this answers is not in the document]
Comment (at-ninja) [reply projects/01_triangle/Triangle.cpp]: Fixed, same as above.

**Comment by apazylbe** [comment]
Checked android build manually, it's marked not required temporarily unrelated to this PR

**Pull request merged**